File: customObjectsComplete/Bicycle.java
/**
 * Bicycle Class
 * @author Sean Morrow
 */
public class Bicycle {

    // Data Member   
    private String ownerName;
    private String model;
    private int ticketCount;
    
    //Constructor
    public Bicycle(String bikeModel) {
        ownerName = "unassigned";
        model = bikeModel;
        ticketCount = 0;
    }
    
    //Returns the name of this bicycle's owner
    public String getOwnerName( ) {
        return ownerName;
    }

    //Assigns the name of this bicycle's owner
    public void setOwnerName(String name) {
        ownerName = name;
    }
    
    public String getModel( ) {
        return model;
    } 

    public int getTicketCount() {
        return ticketCount;
    }
	
    // --------------------------------------------- public methods
    public void violation() {
    	ticketCount++;
    }
           
 }

File: customObjectsComplete/Bicycle.js
/*jshint browser:true */
/*jshint devel:true */

/*
// PROTOTYPE APPROACH TO CUSTOM CLASSES
function Bicycle(bikeModel) {
    // this is our constructor method for Bicycle class
    this.ownerName = "unassigned";
    this.model = bikeModel;
    this.ticketCount = 0;
}

// ------------------------------------ get/set methods
Bicycle.prototype.getOwnerName = function() {
    return this.ownerName;   
};

Bicycle.prototype.setOwnerName = function(name) {
    this.ownerName = name;   
};

Bicycle.prototype.getModel = function() {
    return this.model;   
};

Bicycle.prototype.getTicketCount = function() {
    return this.ticketCount;   
};

// --------------------------------- public methods
Bicycle.prototype.violation = function() {
    this.ticketCount++;   
};
*/

// CLOSURE APPROACH TO CUSTOM CLASSES
function Bicycle(bikeModel) {
	"use strict";
    
    // property variables
    var ownerName = "unassigned";
    var model = bikeModel;
    var ticketCount = 0;
    
    // ------------------------------- get/set methods
    this.getOwnerName = function() {
        return ownerName;   
    };
    
    this.setOwnerName = function(name) {
        ownerName = name;   
    };
    
    this.getModel = function() {
        return model;
    };
    
    this.getTicketCount = function() {
        return ticketCount;   
    };
    
    // ------------------------------ public method
    this.violation = function() {
        ticketCount++;   
    };
    
}




























